src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallRecords/RegistryWorkloadInstallationRecordRepository.cs

```diff
@@ -83,8 +85,44 @@ public IEnumerable<WorkloadId> GetInstalledWorkloads(SdkFeatureBand sdkFeatureBa
         {
             using RegistryKey wrk = _baseKey.OpenSubKey(Path.Combine(BasePath, $"{sdkFeatureBand}"));
 
+            return GetWorkloadInstallationRecordsFromRegistry(wrk);
+        }
+
+        public IEnumerable<WorkloadId> GetInstalledWorkloadsForFeatureBandAndBelow(SdkFeatureBand sdkFeatureBand)
+        {
+            using RegistryKey wrk = _baseKey.OpenSubKey(Path.Combine(BasePath, $"{sdkFeatureBand}"));
+            List<WorkloadId> allWorkloads = new();
+            var sdkBands = wrk?.GetSubKeyNames();
+
+            foreach (var band in sdkBands)
+            {
+                Log?.LogMessage($"Clean --all: Obtaining workload installation records for {BasePath} + {band}.");
```

Comment:
For MSI based installs we already log the initial command and arguments. If we ever renamed `clean --all`, this text would need to be updated. It might also be confusing if we ever used this code for something else in the future.

Reply:
Good point. I've removed all references to the clean command in the log messages.